rascal-vscode-extension/src/lsp/RascalLSPConnection.ts

```diff
@@ -62,8 +62,13 @@ export async function activateLanguageClient(
     const schemesReply = client.sendRequest<string[]>("rascal/filesystem/schemes");
 
     schemesReply.then( schemes => {
-        vfsServer.ignoreSchemes(schemes);
-        new RascalFileSystemProvider(client).registerSchemes(schemes);
+        // avoid registering the same schemes multiple times
+        const knownRascalNativeSchemes = vfsServer.getIgnoredSchemes();
```

Comment:
the function name `getIgnoredSchemes` is confusing. (especially in context to the field & the variable name).

Reply:
Agree. The internal names and the public function `ignoreSchemes` is already awkward.